lib/saluki-components/src/destinations/datadog_metrics/endpoint/endpoints.rs

```diff
@@ -0,0 +1,152 @@
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
+
+use http::Uri;
+use regex::Regex;
+use saluki_error::GenericError;
+use saluki_metadata;
+use serde::Deserialize;
+use snafu::{ResultExt, Snafu};
+use url::Url;
+
+use crate::destinations::datadog_metrics::DEFAULT_SITE;
+
+/// AdditionalEndpoints holds the endpoints and api keys parsed
```

Comment:
Super small nit: I know that the Go style guide pushes to do `// <function/struct name> ....` for comments, but we avoid this in Saluki (and in Rust, in general).

A more idiomatic doc comment for this might be:

```suggestion
/// A set of additional API endpoints to forward metrics to.
///
/// Each endpoint can be associated with multiple API keys. Requests will be forwarded to each unique endpoint/API key pair.
```